compiler/src/main/kotlin/software/amazon/lastmile/kotlin/inject/anvil/processor/GenerateKotlinInjectComponentProcessor.kt

```diff
@@ -115,12 +118,21 @@ internal class GenerateKotlinInjectComponentProcessor(
 
         val isInterface = clazz.classKind == ClassKind.INTERFACE
         val parameters = clazz.primaryConstructor?.parameters ?: emptyList()
+        val componentAnnotations = mutableMapOf<String, TypeName>()
```

Comment:
Keeping state in mutable fields can become problematic. In this case it isn't even needed. I was so free and updated the PR.